build.sh

```diff
@@ -249,25 +249,31 @@ fi
 
 TESTS=0
 BACKEND=0
-CONDA_DEV=0
-CONDA_TEST=0
+BUILD_CONDA_DEV=0
+BUILD_CONDA_TEST=0
+USE_CONDA_DEV=0
```

Comment:
Can't we just assume that if the user passes a tag for these layers, they want to make use of them?

Reply:
My apologies; where I put this comment made it unclear what I was getting at. Let me take another stab at this.

I like the functionality a lot here, but I may be confused about the workflow. It seems like we would want the following behaviors:

- A "complete" build should include building the conda images by default since it does not imply any additional work beyond what we would otherwise do
- If it is not a complete build, we should only _use_ the conda images if the tag has been provided

My understanding right now is that [this line](https://github.com/triton-inference-server/fil_backend/pull/282/files/73d115f1d9f9dddda4b3f1b3a20a46778d0149de#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823R273) means that we have to explicitly opt-in to build the conda images. If we changed that, then [this line](https://github.com/triton-inference-server/fil_backend/pull/282/files/73d115f1d9f9dddda4b3f1b3a20a46778d0149de#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823R170) could become something like
```
if [ -z $CONDA_DEV_TAG ] && !completeBuild
```
and then everything else could just be predicated on the existence of `$CONDA_DEV_TAG`.

It seems like that would help simplify the code for future maintenance by ensuring that we can't _possibly_ have a situation where we're building/using the conda layer but don't have a tag for it. Does that make sense or am I thinking about the workflow wrong?

Reply:
I added `--prebuilt-conda` so that the user has to explicitly opt into using cached Conda envs

Reply:
Sorry, I think I'm still being unclear. The big thing I'm trying to get at is that I think we have too many options here, and some of the redundancy makes it possible to give logically incoherent configurations to the script when I don't think it has to be that way.

Specifically, I _think_ that all that should matter is whether or not `CONDA_DEV_TAG` and/or `CONDA_TEST_TAG` are non-empty variables. If they are, we get one consistent set of behaviors for all other options, and if they are not, we get another consistent set of behaviors. If that's true, it would be ideal to control everything based solely on whether or not those are non-empty. These build scripts tend to accumulate more and more options over time, so if there's a way to control the complexity from the start, it seems like that would be preferable.

Want to ping me offline whenever you're around and we can go through it together? It's still entirely possible there's an aspect of the workflow I'm misunderstanding and we do indeed need all these options/flags.

Reply:
I created `build_conda_env_container.sh` that builds Conda layers. Now `build.sh` has a clear behavior: it will use pre-built Conda layers when `CONDA_DEV_TAG` and/or `CONDA_TEST_TAG` are set.